### DISCUSSION TITLE
Switcher Plus directly supporting inserting link to a heading?

### DISCUSSION THREAD

**Discussion by KennyDenvers** [darlal/obsidian-switcher-plus#107] [Q&A] [answered]
Hi! I enjoy very much the feature of Quick Switcher ++ that I can find and navigate directly to headings in my files. As I was just inserting a link to another note I felt that it would be great if also in doing that (inserting link to another note) it would be fantastic if I would see suggestions for headings that contain the words / characters that I am entering (possibly being triggered by using "#"). Is this feature available through Quick Switcher ++? 
Or does anybody know another way how this is supported? 
(My use case is that I would know that I have a heading in a note that contains - let's say - "abatement", but I can't remember in which file that is. I could go to Quick Switcher ++, look through the "headings search" and then go back inserting the correct file -- but I am wondering if there is an easier way).

**Comment by darlal** [comment]
Hey @KennyDenvers, when you say "inserting a link to another note", I think you're referring to using the `Link Suggester` when you type `[[` and get popup suggestions, is that correct? if so, Switcher++ doesn't currently integrate with that part of the system. Also, after typing `[[` if you immediately type `#` you'll get heading suggestion for the current file - I know this doesn't address your use-case, just mentioning it for completeness.

One thing that might make this workflow a bit smoother for you is feature request #82.  Which would enable you to use the switcher to find a heading with "abatement", then simply using a hotkey to insert a link in the note being edited automatically rather than having to manually type it all. Would that be a bit smoother?

**Reply by KennyDenvers** [reply]
Thank you for your response, @darlal ! Yes, if I understand your interpretation of #82 correctly that would indeed help. Instead of typing [[ I would activate Switcher++ (in headings mode or using #) while editing a note and then use a hotkey (or Alt+Return or something) to insert the link in the currently edited note (instead of opening that other note).
Thank you for your work - and have a great day!

**Comment by darlal** [comment]
Cool. When you have a moment upvote #82 with a 👍 to show interest. It's in the backlog for future implementation. Thanks!

**Reply by darlal** [reply, marked as answer]
Hey @KennyDenvers #82 is live in [release 3.3.3](https://github.com/darlal/obsidian-switcher-plus/releases/tag/3.3.3). Thanks!
